plugins/catalog-backend/config.d.ts

```
@@ -38,6 +38,32 @@ export interface Config {
        * E.g. ["Component", "API", "Template", "Location"]
        */
       allow: Array<string>;
+      /**
+       * Limit this rule to a specific location
+       *
+       * Example with a fixed location
+       *  { "type": "url", "target": "https://github.com/a/b/blob/file.yaml}
+       *
+       * Example using a Regex
+       *  { "type": "url", "match": "https://github.com/a/*}
+       *
+       */
+      location?: Array<{
+        /**
+         * The type of location, e.g. "url".
+         */
+        type: string;
+        /**
+         * The target URL of the location, e.g.
+         * "https://github.com/org/repo/blob/master/users.yaml".
+         */
+        target?: string;
+        /**
+         * The target Regex of the location, e.g.
+         * "https://github.com/org/*.
+         */
+        match?: string;
```

Naming here is tricky. Getting a bit inspired by https://www.envoyproxy.io/docs/envoy/latest/api-v3/type/matcher/v3/string.proto, I'm thinking we could perhaps go with `exact` and `pattern`? Feeling that it's heavily implied that we're referring to the `target` already, since the `type` has to be provided. Otherwise something like `target.exact` and `target.pattern` could work too.

Perhaps it makes sense to validate that exactly one of the options is set too, because being able to set both is just gonna lead to confusion.

---

I hesitated a bit on changing target because that already there but it didn't seem to be in use. I think that exact/pattern sounds good too. 

How would we restrict to setting only one? I just assumed the `exact` would just override the `pattern` field (prioritize the most specific one).

I guess we could make the target take `{ "exact": string } | { "pattern": string}` or something like that but if that is not followed the rule would just be ignored or importing would break altogether. 

I may be missing a good option here, not sure if you had something in mind.

---

`if (l.exact && l.pattern) { throw new Error(...) }` :D

---

Sure, I was hoping to not need to throw an error there. Now changing `target` to `exact` would be breaking change right? I was working on the change and noticed it causes other tests to fail. Is that what we want to do?

---

I made the change. Thanks for the recommendation!